xwiki-platform-core/xwiki-platform-icon/xwiki-platform-icon-fontawesome/src/main/resources/IconThemes/FontAwesome.xml

```diff
@@ -40,8 +40,8 @@
 xwiki.iconset.type = font
 xwiki.iconset.ssx = IconThemes.FontAwesome
 xwiki.iconset.jsx = IconThemes.FontAwesome
-xwiki.iconset.render.wiki = {{html clean="false"}}&lt;span class="fa fa-$icon"&gt;&lt;/span&gt;{{/html}}
-xwiki.iconset.render.html = &lt;span class="fa fa-$icon"&gt;&lt;/span&gt;
+xwiki.iconset.render.wiki = {{html clean="false"}}&lt;span class="fa fa-$icon" role="img"&gt;&lt;/span&gt;{{/html}}
+xwiki.iconset.render.html = &lt;span class="fa fa-$icon" role="img"&gt;&lt;/span&gt;
```

Comment:
IMO this change warrants a comment to explain why we use `role="img"` as it's not something obvious and not a well-known rule, according to https://www.w3.org/WAI/GL/wiki/Providing_a_Semantically_Identified_Icon_Font_with_role%3Dimg 

My worry is how future readers or the code will understand why this `role` attribute was added.

Now I agree that it could add comment duplications so maybe a better solution is to document it on xwiki.org somewhere as a best practice instead. For example under a WCAG section in https://dev.xwiki.org/xwiki/bin/view/Community/CodeStyle/XhtmlCssCodeStyle/ which could link to https://dev.xwiki.org/xwiki/bin/view/Community/WCAGTesting or vice versa.

WDYT?

Reply:
Indeed, an explanation would be helpful, I've added a WCAG section in [the code style page](https://dev.xwiki.org/xwiki/bin/view/Community/CodeStyle/XhtmlCssCodeStyle/), with a link to the WCAG testing page, which itself already contains a link to the code style page, what do you think?